sys/shell/commands/sc_gnrc_netif.c

```diff
@@ -360,10 +398,18 @@ static void _netif_list(kernel_pid_t iface)
         printf(" HWaddr: %s ",
                gnrc_netif_addr_to_str(hwaddr, res, hwaddr_str));
     }
+#ifdef MODULE_LORA
+    uint32_t u32;
+    res = gnrc_netapi_get(iface, NETOPT_CHANNEL, 0, &u32, sizeof(u32));
+    if (res >= 0) {
+        printf(" Channel: %" PRIu32 " ", u32);
+    }
```

Comment:
Soooo... you can't set the channel of an IEEE 802.15.4 device with LoRA anymore now? Because for all the devices that have a 16-bit wide channel as documented in `netopt.h` this will evaluate to something random...

Reply:
So what is the best solution ? Adding another NETOPT_LORA_CHANNEL ? or just use u32 for all devices ?

Reply:
There is the same issue with TX power: SX127x defines an int8_t whereas other radios have int16_t. But it will be easy to use int16_t for the sx127x as well.

Reply:
> So what is the best solution ? Adding another NETOPT_LORA_CHANNEL ? or just use u32 for all devices ?

Why is this value `uint32_t` anyway? Are there really LoRA channels >65535? This is where we should start our discussion on the requirement of an API change ;-).

Reply:
> There is the same issue with TX power: SX127x defines an int8_t whereas other radios have int16_t. But it will be easy to use int16_t for the sx127x as well.

Why is this an issue? Just cast the value to `int8_t` and make sure (with an assert) that the provided value is within the range (`INT8_MIN`, `INT8_MAX`) (both things should of course be happening either in the device driver or within a #7736-style LoRA-layer).

Reply:
> Why is this value uint32_t anyway?

LoRa channels correspond to the center frequency in Hz, e.g 868100000, 868300000 in Europe. That's why uint32_t is required. Maybe we can use some enums and stick to the uint16_t but then it becomes more complex in the driver implementation because regions have different channels values (915000000 for US, etc) and number.

Reply:
Would it work to define this value as the center frequency multiplied by 100Khz? Then the values would be between 8681, 8683 etc?

Reply:
Ok, then since this is different enough from other radio technologies, I would rather vote to introduce another NETOPT value. But not `LORA_CHANNEL`. How about `NETOPT_FREQ_BAND` for (center of a) frequency band?

Reply:
(this way it can also be used by other technologies that use a similar channel scheme)

Reply:
> How about NETOPT_FREQ_BAND 

+1, I already started to do this change ;)

Reply:
I finally opted for NETOPT_CHANNEL_FREQUENCY which is more IMHO in the spirit of LoRa technology. The band is more for 868MHz, 915MHz, 433MHz, etc. Each band is divided into channels of different widths (125kHz, 250kHz and 500kHz)

Reply:
@bergzand

> Would it work to define this value as the center frequency multiplied by 100Khz? Then the values would be between 8681, 8683 etc?

I think this would be more confusing than having just a separate option.

Reply:
> I think this would be more confusing than having just a separate option.

Yeah, I was trying to fit it into the existing netopt here, I agree with introducing a new netopt value for this, probably more clear than trying to multiplex different interpretatioins over a single netopt. :)